src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -0,0 +1,278 @@
+        /// <param name="value">The value to be written as a JSON string as part of the name/value pair.</param>
+        /// <param name="suppressEscaping">If this is set, the writer assumes the property name is properly escaped and skips the escaping step.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the specified property name is too large.
+        /// </exception>
+        /// <exception cref="JsonWriterException">
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WriteString(ReadOnlySpan<byte> propertyName, DateTime value, bool suppressEscaping = false)
+        {
+            JsonWriterHelper.ValidateProperty(ref propertyName);
+
+            if (!suppressEscaping)
+                WriteStringSuppressFalse(ref propertyName, value);
+            else
+                WriteStringByOptions(ref propertyName, value);
+
+            _currentDepth |= 1 << 31;
+            _tokenType = JsonTokenType.String;
+        }
+
+        private void WriteStringSuppressFalse(ref ReadOnlySpan<char> propertyName, DateTime value)
+        {
+            int propertyIdx = JsonWriterHelper.NeedsEscaping(propertyName);
+
+            Debug.Assert(propertyIdx >= -1 && propertyIdx < int.MaxValue / 2);
+
+            if (propertyIdx != -1)
+            {
+                WriteStringEscapeProperty(ref propertyName, value, propertyIdx);
+            }
+            else
+            {
+                WriteStringByOptions(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringSuppressFalse(ref ReadOnlySpan<byte> propertyName, DateTime value)
+        {
+            int propertyIdx = JsonWriterHelper.NeedsEscaping(propertyName);
+
+            Debug.Assert(propertyIdx >= -1 && propertyIdx < int.MaxValue / 2);
+
+            if (propertyIdx != -1)
+            {
+                WriteStringEscapeProperty(ref propertyName, value, propertyIdx);
+            }
+            else
+            {
+                WriteStringByOptions(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringEscapeProperty(ref ReadOnlySpan<char> propertyName, DateTime value, int firstEscapeIndexProp)
+        {
+            Debug.Assert(int.MaxValue / MaxExpansionFactorWhileEscaping >= propertyName.Length);
+
+            char[] propertyArray = null;
+
+            int length = firstEscapeIndexProp + MaxExpansionFactorWhileEscaping * (propertyName.Length - firstEscapeIndexProp);
+            Span<char> span;
+            if (length > StackallocThreshold)
+            {
+                propertyArray = ArrayPool<char>.Shared.Rent(length);
+                span = propertyArray;
+            }
+            else
+            {
+                // Cannot create a span directly since the span gets exposed outside this method.
```

Comment:
The compiler won't let me pass a stackalloc span to a method (even if it is never returned).

Reply:
Sure it will... just not when you're passing around another ref struct by ref and letting it escape.  This is one of the reasons I really don't like the use of `ref Readonlyspan` that you've got happening.

Reply:
> Sure it will...

I tried after fixing passing spans by ref (in most cases I changed to in, and in places where we are using array pool, I am doing regular pass by value). 

See https://github.com/dotnet/corefx/pull/34425#discussion_r247299400